src/components/Table/BaseTable.tsx

```diff
@@ -17,56 +17,78 @@ type SpruceTableProps = {
-              </HeaderCell>
+  <Table data-cy={dataCyTable} table={table} {...args}>
+    <TableHead>
+      {table.getHeaderGroups().map((headerGroup) => (
+        <HeaderRow key={headerGroup.id}>
+          {headerGroup.headers.map((header) => (
+            <HeaderCell key={header.id} header={header}>
+              {flexRender(header.column.columnDef.header, header.getContext())}
+              {/* @ts-ignore-error */}
+              {header.column.columnDef?.meta?.filterComponent?.({
+                column: header.column,
+              })}
+              {/* @ts-ignore-error */}
+              {header.column.columnDef?.meta?.sortComponent?.({
+                column: header.column,
+              })}
+            </HeaderCell>
+          ))}
+        </HeaderRow>
+      ))}
+    </TableHead>
+    <TableBody>
+      {table.getRowModel().rows.map((row) => (
+        <Row
+          key={row.id}
+          row={row}
+          data-cy={dataCyRow}
+          className={css`
+            &[aria-hidden="false"] td > div {
+              max-height: unset;
+            }
+          `}
+        >
+          {row.getVisibleCells().map((cell) => (
+            <Cell key={cell.id}>
+              {flexRender(cell.column.columnDef.cell, cell.getContext())}
+            </Cell>
+          ))}
+          {row.original.renderExpandedContent && <ExpandedContent row={row} />}
+          {row.subRows &&
+            row.subRows.map((subRow) => (
+              <Row
+                key={subRow.id}
+                row={subRow}
+                className={css`
+                  &[aria-hidden="false"] td > div[data-state="entered"] {
+                    max-height: unset;
+                  }
+                `}
+              >
+                {subRow.getVisibleCells().map((cell) => (
+                  <Cell key={cell.id}>
+                    {flexRender(cell.column.columnDef.cell, cell.getContext())}
+                  </Cell>
+                ))}
+              </Row>
             ))}
-          </HeaderRow>
-        ))}
-      </TableHead>
-      <TableBody>
-        {table.getRowModel().rows.map((row) => (
-          <Row
-            key={row.id}
-            row={row}
-            data-cy={dataCyRow}
-            className={css`
-              &[aria-hidden="false"] td > div {
-                max-height: unset;
-              }
-            `}
-          >
-            {row.getVisibleCells().map((cell) => (
-              <Cell key={cell.id}>
-                {flexRender(cell.column.columnDef.cell, cell.getContext())}
-              </Cell>
-            ))}
-            {row.original.renderExpandedContent && (
-              <ExpandedContent row={row} />
-            )}
-          </Row>
-        ))}
-      </TableBody>
-    </Table>
-    {table.getRowModel().rows.length === 0 &&
+        </Row>
+      ))}
+    </TableBody>
+    {!loading &&
+      table.getRowModel().rows.length === 0 &&
       (emptyComponent || "No data to display")}
```

Comment:
Would it be possible to center the placeholder 😳 
<img width="1383" alt="image" src="https://github.com/evergreen-ci/spruce/assets/9298431/cf5a9097-e8a1-4b9b-ae69-b92a3ed19dc3">